client/signup/config/flows.js

```diff
@@ -238,11 +239,72 @@ const Flows = {
 			flow = filterDesignTypeInFlow( flow );
 		}
 
-		return flow;
+		return Flows.getABTestFilteredFlow( flowName, flow );
 	},
 
 	getFlows() {
 		return flows;
+	},
+
+	/**
+	 * Preload AB Test variations after a certain step has been completed.
+	 *
+	 * This gives the option to set the AB variation as late as possible in the
+	 * signup flow.
+	 *
+	 * Currently only the `main` flow is whitelisted.
+	 *
+	 * @param {String} flowName The current flow
+	 * @param {String} stepName The step that is being completed right now
+	 */
+	preloadABTestVariationsForStep( flowName, stepName ) {
```

Comment:
> It does make sense to do it in one place :) I updated the code to do that, now the preloadABTestVariationsForStep is being called from within getFlow, so everything is in one place.

My point was that you don't need both of those methods. You can remove `Flows.preloadABTestVariationsForStep()` and just put your filtering logic (am I in "main" flow and "survey" step?) in `Flows.getABTestFilteredFlow()`. You don't need to assign the variant separately from checking it. `abtest()` returns the variant, so you shouldn't typically need to use `getABTestVariation()`.